Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -109,7 +111,13 @@ public void UpdateUserInterface(EntityUid? user, EntityUid store, StoreComponent
 
         // only tell operatives to lock their uplink if it can be locked
         var showFooter = HasComp<RingerUplinkComponent>(store);
-        var state = new StoreUpdateState(component.LastAvailableListings, allCurrency, showFooter, component.RefundAllowed);
+
+        var discounts = Array.Empty<StoreDiscountData>();
+        if (TryComp<StoreDiscountComponent>(store, out var discountsComponent))
+        {
+            discounts = discountsComponent.Discounts;
+        }
+        var state = new StoreUpdateState(component.LastAvailableListings, allCurrency, discounts, showFooter, component.RefundAllowed);
```

Comment:
You added a separate system but then hardcoded this cyclic dependency in anyway.

Just make it raise an event instead.

Reply:
de-coupled systems using events. Using 'GetXEvent' with mutable object feels wierd, tbh. Maybe i was wrong to try and decouple them in the first place, but StoreSystem is so big ;(